blocks_vertical/procedures.js

```diff
@@ -38,28 +34,28 @@ goog.require('Blockly.ScratchBlocks.VerticalExtensions');
  * @return {!Element} XML storage element.
  * @this Blockly.Block
  */
-Blockly.ScratchBlocks.ProcedureUtils.callerMutationToDom = function() {
+function callerMutationToDom() {
   var container = document.createElement('mutation');
   container.setAttribute('proccode', this.procCode_);
   container.setAttribute('argumentids', JSON.stringify(this.argumentIds_));
   container.setAttribute('warp', JSON.stringify(this.warp_));
   return container;
-};
+}
 
 /**
  * Parse XML to restore the (non-editable) name and arguments of a procedure
  * call block.
  * @param {!Element} xmlElement XML storage element.
  * @this Blockly.Block
  */
-Blockly.ScratchBlocks.ProcedureUtils.callerDomToMutation = function(xmlElement) {
+function callerDomToMutation(xmlElement) {
```

Comment:
Hmm I just realized we might have an issue with new features only being supported in the JSON serialization. We've been pushing partners to upgrade to that since we don't want to be supporting two different serialization systems forever. You and I should probably have a chat about what that would look like for scratch.

My first thought is that we'll run into trouble with icons because they only support JSON. I'd need to think longer  for other edge cases.

Reply:
Good point, agreed we should talk through this!